### PR TITLE
test coverage for internal/utils

### PR DESCRIPTION
**Pull request by gflarity**
#### What type of PR is this?

Testing

#### What this PR does / why we need it:

Increased test coverage for internal/utils tree.

#### Special notes for your reviewer:

Prompt:

Please create an idiomatic golang unit tests for this specific directory/package. Herea are the instructions: 

* Start by revieiwng the state of test coverage with golang tooling. 
* Be sure to document the fields in the anonymous test structs and how they'll be used below. 
* Each Test* function should also have some concise documentation as well. Each test case should have a brief explanation and expectation explaine. This should be right before the go struct definition for that test if it's table driven tests. 
* Each test should have a name field for cross referencing failures. 
* Rather than having a description field, the description should just be in comments above the name in the struct declaration. 
* Don't forget to run the tests and confirm they're working, but focus on just testing this package as others might have issues. 
* Please avoid creating skipped tests. 
* Refactor code to make it more testable if necessary. However keep these refactors to a minimum and as simple as possible. 
* Specifically hardcoded constants etc are good candidates to be refactor out of testable code. As are interface that can be mocked in a straight forward way. However you must remember to ensure backwards compatibility and new functions rather than changing the signature of public functions. Avoid making these test helpers public though. Only make such refactors when there's a good return on investment. 
* Please ensure coverage is as good as possible using golang tooling. 
* Avoid creating table driven tests when the test function just has a single test in it. 
* Avoid test that don't really test anything. 
* Avoid describing tests as edge cases. 
* Avoid tests for command line arguments. Try to keep test cases unique and adding value (avoid duplicate test cases). 

#### Does this PR introduce a API change?
```release-note
None
```

### CODE REVIEW

File: operator/internal/utils/concurrent_test.go [hunk 1]
```diff
@@ -206,3 +206,48 @@ func createErringTaskFn(name string) func(ctx context.Context) error {
...
+	assert.NotNil(t, aggregatedErr)
+	assert.Contains(t, aggregatedErr.Error(), "task failed")
+
+	// Multiple errors returns joined error
+	err2 := fmt.Errorf("another failure")
+	result = RunResult{Errors: []error{err1, err2}}
+	aggregatedErr = result.GetAggregatedError()
+	assert.NotNil(t, aggregatedErr)
+	assert.Contains(t, aggregatedErr.Error(), "task failed")
+	assert.Contains(t, aggregatedErr.Error(), "another failure")
```
Comment (renormalize) [inline operator/internal/utils/concurrent_test.go]:
```suggestion
	assert.True(t, errors.Is(aggregatedErr, err1))
	assert.True(t, errors.Is(aggregatedErr, err2))
```

File: operator/internal/utils/concurrent_test.go [hunk 2]
```diff
@@ -206,3 +206,48 @@ func createErringTaskFn(name string) func(ctx context.Context) error {
...
+		SuccessfulTasks: []string{"task1", "task2"},
+		FailedTasks:     []string{"task3"},
+		SkippedTasks:    []string{"task4", "task5"},
+	}
+	summary = result.GetSummary()
+	assert.Contains(t, summary, "task1")
+	assert.Contains(t, summary, "task2")
+	assert.Contains(t, summary, "task3")
+	assert.Contains(t, summary, "task4")
+	assert.Contains(t, summary, "task5")
```
Comment (renormalize) [inline operator/internal/utils/concurrent_test.go]: Establishes the task exists in the summary, along with the categorization; instead of just the former in the current test. Also is consistent with the empty result test above.

```suggestion
	assert.Contains(t, summary, "SuccessfulTasks: [task1 task2]")
	assert.Contains(t, summary, "FailedTasks: [task3]")
	assert.Contains(t, summary, "SkippedTasks: [task4 task5]")
```

File: operator/internal/utils/concurrent_test.go [hunk 3]
```diff
@@ -206,3 +206,48 @@ func createErringTaskFn(name string) func(ctx context.Context) error {
...
+	// No errors returns nil
+	result := RunResult{Errors: []error{}}
+	assert.Nil(t, result.GetAggregatedError())
+
+	// Single error returns that error
+	err1 := fmt.Errorf("task failed")
+	result = RunResult{Errors: []error{err1}}
+	aggregatedErr := result.GetAggregatedError()
+	assert.NotNil(t, aggregatedErr)
+	assert.Contains(t, aggregatedErr.Error(), "task failed")
```
Comment (renormalize) [inline operator/internal/utils/concurrent_test.go]: Since the errors are `Join()`ed, I think this is a better way to verify things.

```suggestion
	assert.True(t, errors.Is(aggregatedErr, err1))
```

File: operator/internal/utils/concurrent_test.go (hunk 1 above)
Comment (renormalize) [inline operator/internal/utils/concurrent_test.go]:
```suggestion
	assert.True(t, errors.Is(aggregatedErr, err1))
	assert.True(t, errors.Is(aggregatedErr, err2))
```

File: operator/internal/utils/kubernetes/pod_test.go [hunk 4]
```diff
@@ -113,3 +115,552 @@ func TestIsPodScheduled(t *testing.T) {
...
+
+	// Check started but not ready pods
+	assert.Len(t, categories[PodStartedButNotReady], 1)
+	assert.Equal(t, "started-not-ready-pod", categories[PodStartedButNotReady][0].Name)
+}
+
+// Helper function to create a bool pointer
+func boolPtr(b bool) *bool {
+	return &b
+}
```
Comment (renormalize) [inline operator/internal/utils/kubernetes/pod_test.go]: As mentioned in https://github.com/ai-dynamo/grove/pull/230#discussion_r2499400091, let's use the generic function `ptr.To()`.

**Review by renormalize — CHANGES_REQUESTED** [review]
thanks for the PR @gflarity!
